File: Python/TreeSort.py
from random import randint

class Node():
    def __init__(self,value):
        self.value=value
        self.right="None"
        self.left="None"

class BinaryTree():
    root="None"
    def AddNode(self,value,curNode):
        if self.root=="None":
            self.root=Node(value)
            curNode=self.root
            return "pass"
            print(curNode.value)
        
        elif curNode.value>=value:
            if curNode.left=="None":
                curNode.left=Node(value)
            else:
                self.AddNode(value,curNode.left)
        
        else:
            if curNode.right=="None":
                curNode.right=Node(value)
            else:
                self.AddNode(value,curNode.right)
    
    def inOrder(self,curNode):
        l=[]
        if curNode.left!="None":
            self.inOrder(curNode.left)
            l.extend(self.inOrder(curNode.left))
        
        l.append(curNode.value)
        #print(curNode.value)
        
        if curNode.right!="None":
            self.inOrder(curNode.right)
            l.extend(self.inOrder(curNode.right))
        
        return l
        
    def FillTree(self):
        for a in range(10):
            self.AddNode(randint(10,10000),self.root)
            
    def TreeSort(self,l):
        for a in l:
            self.AddNode(a,self.root)
        return self.inOrder(self.root)
    
    
tree=BinaryTree()
l=[23,41,23,65,76,46,97,53]
l=tree.TreeSort(l)
print(l)
        
        
        
        
        

File: Js/TreeSort.js
class NodeTree {
  root = null;
  left = null;
  right = null;
  constructor(rootValue) {
    this.root = rootValue;
  }
}
class Tree {
  root = null;
  addNode = (value, curNode = this.root) => {
    if (this.root === null) {
      this.root = new NodeTree(value);
    } else if (value > curNode.root) {
      if (curNode.right === null) {
        curNode.right = new NodeTree(value);
      } else {
        return this.addNode(value, curNode.right);
      }
    } else {
      if (curNode.left === null) {
        curNode.left = new NodeTree(value);
      } else {
        return this.addNode(value, curNode.left);
      }
    }
  };
  inOrder = (curNode = this.root) => {
    let l = [];
    if (curNode !== null && curNode.left !== null) {
      l.push(...this.inOrder(curNode.left));
    }
    l.push(curNode.root);
    if (curNode !== null && curNode.right !== null) {
      l.push(...this.inOrder(curNode.right));
    }
    return l;
  };
  sort = (l) => {
    l.forEach((e) => this.addNode(e));
    return this.inOrder();
  };
}

const tree = new Tree();
const l = [2, 5, 1, 7, 3, 9, 43, 12, 13, 13, 54, 23, 11];
console.log(tree.sort(l));
